### PR TITLE
refactor: remove useless spread

### PR DESCRIPTION
**Pull request by exoego**
### The author should do the following, if applicable

- [ ] Add tests
- [x] Run tests
- [x] `bun run format:fix && bun run lint:fix` to format the code
- [ ] Add [TSDoc](https://tsdoc.org/)/[JSDoc](https://jsdoc.app/about-getting-started) to document the code

### CODE REVIEW

File: src/client/client.ts [hunk 1]
```diff
@@ -80,7 +80,7 @@ class ClientRequestImpl {
     let methodUpperCase = this.method.toUpperCase()
 
     const headerValues: Record<string, string> = {
-      ...(args?.header ?? {}),
+      ...args?.header,
```
Comment (exoego) [inline src/client/client.ts:83]:
```typescript
const a = undefined
{ ...a, b: 1 }
// { b: 1}
```

Comment (EdamAme-x) [reply src/client/client.ts:83]: I put out a PR related to this.

https://github.com/exoego/hono/pull/5

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@exoego @EdamAme-x 

Thank you both!

**Pull request merged**